src/module_2/groceries_analysis.md

```diff
@@ -0,0 +1,1811 @@
+
+plt.bar(common_items['variant_id'][:10].astype(str), common_items['revenue'][:10], edgecolor='k')
+plt.title('Top 10 Items by Revenue')
+plt.xlabel('Item')
+plt.ylabel('Revenue')
+plt.xticks(ticks=range(0,10),labels=common_items['product_type'][:10],rotation=45, ha='right')
+plt.show()
+
+
+```
+
+
+    
+![png](groceries_analysis_files/groceries_analysis_17_0.png)
+    
+
+
+As we can see, the product which produces the most revenue by far, is part of the milk category. This might indicate that the platform relies too much in selling this kind of milk-related product. To check this, let's compare the revenue of the top ten products against the revenue of the rest of the products.
+
+
+```python
+# Calculate the total revenue of the rest of the dataset
+rest_revenue = common_items['revenue'][10:].sum()
+
+# Create a new dataframe to hold the data for the bar plot
+revenue_data = pd.DataFrame({'Product': ['Top 10 Items', 'Rest of Dataset'],
+                             'Revenue': [common_items['revenue'][:10].sum(), rest_revenue]})
+
+# Plot the bar chart
+plt.bar(revenue_data['Product'], revenue_data['Revenue'], edgecolor='k')
+plt.title('Revenue Comparison')
+plt.xlabel('Product')
+plt.ylabel('Revenue')
+plt.show()
+
+# Calculate percentage of revenue from the top 10 items
+top_10_revenue = common_items['revenue'][:10].sum()
+total_revenue = common_items['revenue'].sum()
+top_10_revenue_percentage = top_10_revenue / total_revenue * 100
+printmd(f"Percentage of revenue from the top 10 items: {top_10_revenue_percentage:.2f}%")
+# Calculate percentage of revenue from the rest of the dataset
+rest_revenue = common_items['revenue'][10:].sum()
+rest_revenue_percentage = rest_revenue / total_revenue * 100
+printmd(f"Percentage of revenue from the rest of the dataset: {rest_revenue_percentage:.2f}%")
+# Calculate percentage of revenue from the top product 
+top_product_revenue = common_items['revenue'][0]
+top_product_revenue_percentage = top_product_revenue / total_revenue * 100
+printmd(f"Percentage of revenue from the top product: {top_product_revenue_percentage:.2f}%")
+
+```
+
+
+    
+![png](groceries_analysis_files/groceries_analysis_19_0.png)
+    
+
+
+
+Percentage of revenue from the top 10 items: 20.68%
+
+
+
+Percentage of revenue from the rest of the dataset: 79.32%
+
+
+
+Percentage of revenue from the top product: 9.88%
+
+
+The platform's product portfolio doesn't appear to be well balanced in terms of its item revenues, with the top ten products accounting for 20% of the total revenue, while the other 1467 products account for the remaining 80% . Moreover, the top product represents 10% of the total revenue, which is a lot for a single product. Let's now check revenue by general categories.
+
+
+```python
+category_revenue = common_items.groupby('product_type')['revenue'].sum().reset_index().sort_values('revenue', ascending=False)
+fig, ax = plt.subplots(1,2, figsize=(15,5))
```

Comment:
What is the rationale of here using the fig, ax and before only plt? As I said I prefer this approach, but lack of consistency is not a good practice to follow the code.

Reply:
Basically, if I'm creating just one plot, I tend to use plt because I don't need to discriminate between plot configurations, however, if I'm creating multiple subplots(2 or more) I use ax because i want to customize the content of each plot I display.